src/taipy/core/_scheduler/_scheduler.py

```diff
@@ -200,11 +209,22 @@ def __remove_jobs_to_run(cls, jobs):
         cls.jobs_to_run = new_jobs_to_run
 
     @classmethod
-    def _cancel_jobs(cls, job_id_to_cancel, jobs):
+    def _fail_subsequent_jobs(cls, failed_job: Job):
+        with cls.lock:
+            to_fail_jobs = set()
```

Comment:
I would rename it to 'to_abandon_jobs'

Reply:
I'll also add the failed_jobs to the set and rename it to to_fail_or_abandon_jobs :D